community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1793,6 +1811,50 @@ def nodeset_reservation(self, nodeset: object) -> Optional[ReservationDetails]:
             name=name,
             policies=policies,
             bulk_insert_name=nodeset.reservation_name)
+    
+    def future_reservation(self, nodeset:object) -> Optional[FutureReservation]:
+        if not nodeset.future_reservation:
+            return None
+
+        zones = list(nodeset.zone_policy_allow or [])
+        assert len(zones) == 1, "Only single zone is supported if using a reservation"
+        zone = zones[0]
+        
+        project = self.project
+        name = nodeset.future_reservation
+        active = None
+        specific = None
+        active_reservation = None
+        end_time = None
+        start_time = None
+        
+        if match := re.search(r'/futureReservations/(?P<name>[^/]+)(/.*)?$', nodeset.future_reservation):
+            name = match.group("name")
+
+        fr = self._get_future_reservation(project, zone, name)
+
+        start_time = datetime.fromisoformat(fr["timeWindow"]["startTime"][:-1])
+        end_time = datetime.fromisoformat(fr["timeWindow"]["endTime"][:-1])
+
+        if "autoCreatedReservations" in fr["status"] and (fr_res:=fr["status"]["autoCreatedReservations"][0]):
+            if (match := re.search(r'/reservations/(?P<name>[^/]+)(/.*)?$',fr_res)) and (res := self._get_reservation(project, zone, match.group('name'))) and (start_time<=datetime.utcnow()<=end_time):
+                active_reservation = ReservationDetails(project=project,zone=zone,name=res["name"],policies=[],bulk_insert_name=res["name"])
```

Comment:
Could you pleas share (DM me, not here) the example of `res`, my concern is using it as `bulk_insert_name`

Reply:
My lazy recommendation would be to fetch actual reservation from API, create a function shared with `def nodeset_reservation` to make request and compose `ReservationDetails`

Reply:
I will send you this as requested, let's sync offline about this point

Reply:
Added shared function